include/nix/Platform.hpp

```diff
@@ -43,4 +43,8 @@
 #define NIX_SRC_FUNC  __PRETTY_FUNCTION__
 #else
 #define NIX_SRC_FUNC __FUNCSIG__
-#endif
\ No newline at end of file
+#endif
+
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
```

Comment:
that's quite a different solution from what I had in mind yesterday night. Good, that you did it :)